src/Compiler/CompilerUtils.cpp

```diff
@@ -383,7 +389,8 @@ static std::string genSharedLib(string outputBaseName, std::vector<string> opts,
 #else
   string sharedLibPath = outputBaseName + ".so";
   std::vector<string> outputOpt = {"-o", sharedLibPath};
-  std::vector<string> sharedLibOpts = {"-shared", "-fPIC"};
+  std::vector<string> sharedLibOpts = {
+      "-shared", "-fPIC", "-Wl,-rpath=" + getRuntimeDir()};
```

Comment:
The hardcore link has been replaced by getRuntimeDir().

Reply:
As mentioned before, it's typically not a good idea to embed shared library search path inside an executable through rpath since it only resolves the path at compile time. Even if you use `getRuntimeDir()` to replace a hard coded path, it doesn't change the fact that whatever directory returned by `getRuntimeDir()` is still hard coded into the executable and only works in our specific environment. At runtime, if shared libraries are installed into non-standard custom locations, you still need to set `LD_LIBRARY_PATH`. So `LD_LIBRARY_PATH` is the preferred way of specifying shared library search path at runtime. For whichever tests that are failing without rpath, you really should find out where the `LD_LIBRARY_PATH` needs to be set to fix them.